### PR TITLE
Add formula for the ky_taxable_income variable

### LINKED ISSUE #3304: Add formula for the ky_taxable_income variable
Kentucky taxable income is currently not defined.

### PR DESCRIPTION
**Pull request by YichunRen**
### Summary
:sparkles::page_facing_up::us:

Add formula for `ky_taxable_income` variable. This change implements the state income tax calculation for Kentucky based on the official tax tables and brackets.

> _`ky_taxable_income`_
> _A new formula added_
> _Springtime for taxes_

### Walkthrough
*  Add a formula for `ky_taxable_income` based on the state tax brackets and tables ([link](https://github.com/PolicyEngine/policyengine-us/pull/3343/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L1-R3),                            F1L30R

### CODE REVIEW

**Comment by YichunRen** [comment]
Fixes https://github.com/PolicyEngine/policyengine-us/issues/3304

File: policyengine_us/variables/gov/states/ky/tax/income/deductions/ky_itemized_deduction.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from policyengine_us.model_api import *
+
+
+class ky_itemized_deduction(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ky/tax/income/deductions/ky_itemized_deduction.py:4]:
```suggestion
class ky_itemized_deductions(Variable):
```

File: policyengine_us/variables/gov/states/ky/tax/income/deductions/ky_itemized_deduction.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Kentucky itemized deduction"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ky/tax/income/deductions/ky_itemized_deduction.py:7]:
```suggestion
    label = "Kentucky itemized deductions"
```

File: policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py [hunk 3]
```diff
@@ -8,3 +8,18 @@ class ky_taxable_income(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.KY
+    reference = "https://revenue.ky.gov/Forms/740%20Packet%20Instructions%205-9-23.pdf#page=11"
+
+    def formula(tax_unit, period, parameters):
+        ky_agi = tax_unit("ky_agi", period)
+        # current model does not have itemized deduction --> create a dummy one
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py:15]: As long as we have an issue filed we don't need it in the code.
```suggestion
```

File: policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py [hunk 4]
```diff
@@ -8,3 +8,18 @@ class ky_taxable_income(Variable):
...
+        standard_deduction = tax_unit("ky_standard_deduction", period)
+        itemized_deduction = tax_unit("ky_itemized_deduction", period)
+        # if filler itemized deductions exceed standard deductions, it will benefit filler to itemize.
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py:18]:
```suggestion
        # Assume that filers itemize if itemized deductions exceed the standard deduction.
        # They do not need to follow their federal itemization choice.
```

File: policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py [hunk 5]
```diff
@@ -8,3 +8,18 @@ class ky_taxable_income(Variable):
...
+        deduction = where(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ky/tax/income/ky_taxable_income.py:19]:
```suggestion
        deduction = max_(standard_deduction, itemized_deductions)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request closed without merging**